suite/tests/linux/syscall_pwait.c

```diff
@@ -64,24 +93,52 @@ signal_handler(int sig, siginfo_t *siginfo, void *context)
 }
 
 bool
-kick_off_child_signals()
+kick_off_child_usr1()
 {
-    /* waste some time */
-    nanosleep(&sleeptime, NULL);
+    int pid = fork();
+    if (pid < 0) {
+        perror("fork error");
+    } else if (pid == 0) {
+        kill(getppid(), SIGUSR1);
+        *ready_to_listen = true;
```

Comment:
How do you know this store won't be moved above the kill()?  Again, it seems best to use higher-level abstractions: rolling your own synch makes it just too easy to make errors and makes it harder to read and understand the code with its extra complexity.

Reply:
The compiler wouldn't move a store across a system call but we have talked about this already (adhere to Google standard). In any case, the test has been changed to using pthreads and cond vars.